torch/ao/quantization/fx/convert.py

```diff
@@ -131,11 +131,13 @@ def _replace_observer_with_quantize_dequantize_node_decomposed(
     # 1. extract the information from activation_post_process module for generating
     # the quantize and dequantize operator
     dtype = activation_post_process.dtype  # type: ignore[attr-defined]
-    compute_dtype = None
-    if hasattr(activation_post_process, "compute_dtype"):
-        compute_dtype = activation_post_process.compute_dtype  # type: ignore[attr-defined]
+
+    is_dynamic = False
+    if hasattr(activation_post_process, "is_dynamic"):
+        is_dynamic = activation_post_process.is_dynamic  # type: ignore[assignment]
+
     if dtype in [torch.quint8, torch.qint8, torch.qint32] and \
-            not hasattr(activation_post_process, 'compute_dtype'):
+            (not is_dynamic):
```

Comment:
nit (can be done in a separate PR): I think the overall checking can be made clearer as well for this code, right now it is:
```
if dtype in [torch.quint8, torch.qint8, torch.qint32] and not is_dynamic:
   ...
elif is_dynamic:
   ...
else:
   # fp16
```

I think a clearer one should be:
```
if is_dynamic:
   ...
else:
   if dtype in [torch.quint8, torch.qint8, torch.qint32]:
      ...
    else:
      # fp16
      ...
```

Reply:
also we might want to add dtype check for is_dynamic branch as well, looks like we only support uint8, int8, fp16 in the non-decomposed function and uint8 and int8 in the decomposed function